src/parser.rs

```diff
@@ -345,9 +349,14 @@ impl<'i: 't, 't> Parser<'i, 't> {
     pub fn expect_exhausted(&mut self) -> Result<(), BasicParseError<'i>> {
         let start = self.state();
         let result = match self.next() {
-            Err(BasicParseError { kind: BasicParseErrorKind::EndOfInput, .. }) => Ok(()),
+            Err(BasicParseError {
+                kind: BasicParseErrorKind::EndOfInput,
+                ..
+            }) => Ok(()),
             Err(e) => unreachable!("Unexpected error encountered: {:?}", e),
-            Ok(t) => Err(start.source_location().new_basic_unexpected_token_error(t.clone())),
+            Ok(t) => Err(start
```

Comment:
This does look a bit weird as well.

Reply:
Hmm, that's weird indeed. Putting them back on a single line fits under the line width limit too, so I'm not sure why this was formatted. Maybe another bug?
I'll file an issue for the macro bug (couldn't reply to that one for some reason).